### PR TITLE
Rename things in imperative

### LINKED ISSUE #1091: Replace from_imperative with from_delayed?
We recently renamed `do` to `delayed`.  In an effort to reduce names should we replace occurrences of `imperative` with `delayed` as well?  In particular I'm considering the `Collection.from_imperative` functions.  Currently we're asking users to know that the terms `delayed` and `imperative` are related.

### PR DESCRIPTION
**Pull request by jcrist**
This is a different take on commit 21f0aacb7094e205250c72cbc14432a824050eef in #1110. Supersedes #1110, Fixes #1091.

`value` -> `delayed`
`Value` -> `Delayed`
`imperative.py` -> `delayed.py`

### CODE REVIEW

File: dask/delayed.py [hunk 1]
```diff
@@ -262,110 +296,58 @@ def __getattr__(self, attr):
...
-    [1, 2, 3, 1, 2]
-    >>> a[1].compute()
-    2
+for op in [operator.abs, operator.neg, operator.pos, operator.invert,
+           operator.add, operator.sub, operator.mul, operator.floordiv,
+           operator.truediv, operator.mod, operator.pow, operator.and_,
+           operator.or_, operator.xor, operator.lshift, operator.rshift,
+           operator.eq, operator.ge, operator.gt, operator.ne, operator.le,
+           operator.lt, operator.getitem]:
+    Delayed._bind_operator(op)
```
Comment (mrocklin) [inline dask/delayed.py]: Ah, that's much more convenient.

**Comment by mrocklin** [comment]
At first glance it seems that we've removed a class at the cost of adding flags to an existing one.  It's not clear to me that either one is simpler.

**Comment by jcrist** [comment]
I like this one a lot better. It works the same way that we already handle method access on objects - the flag is just to special case root objects when called. It also means that callable objects won't get specialized to one type over the other.

**Comment by mrocklin** [comment]
OK, my uncertainty is pretty high so I'm quite happy to defer here.

**Pull request merged**